File: .claude/skills/fixtures/core6-mechanical/go/rename/before.go
package main

import "fmt"

// ProcessRequest handles an incoming request string and returns a response.
func ProcessRequest(req string) string {
	return fmt.Sprintf("processed: %s", req)
}

func main() {
	r1 := ProcessRequest("hello")
	r2 := ProcessRequest("world")
	fmt.Println(r1, r2)
}

File: .claude/skills/fixtures/core6-mechanical/rust/rename/before.rs
/// process_request handles an incoming request string and returns a response.
/// NOTE: This is NOT a trait method — ast-grep rename is safe here.
fn process_request(req: &str) -> String {
    format!("processed: {}", req)
}

fn main() {
    let r1 = process_request("hello");
    let r2 = process_request("world");
    println!("{} {}", r1, r2);
}
